[BUG] CI failure in Build and Test k-NN Plugin on MacOS

**What is the bug?**
```
[  4%] Building CXX object external/faiss/faiss/CMakeFiles/faiss.dir/AutoTune.cpp.o
In file included from /Users/runner/work/k-NN/k-NN/jni/external/faiss/faiss/AutoTune.cpp:25:
In file included from /Users/runner/work/k-NN/k-NN/jni/external/faiss/faiss/IndexHNSW.h:17:
/Users/runner/work/k-NN/k-NN/jni/external/faiss/faiss/impl/HNSW.h:14:10: fatal error: 'omp.h' file not found
#include <omp.h>
         ^~~~~~~
1 error generated.
make[3]: *** [external/faiss/faiss/CMakeFiles/faiss.dir/AutoTune.cpp.o] Error 1
make[2]: *** [external/faiss/faiss/CMakeFiles/faiss.dir/all] Error 2
make[1]: *** [CMakeFiles/opensearchknn_faiss.dir/rule] Error 2
make: *** [opensearchknn_faiss] Error 2

Task :buildJniLib FAILED
FAILURE: Build failed with an exception.
```
**How can one reproduce the bug?**
Failure example: [Build and Test k-NN Plugin on MacOS (11)](https://github.com/opensearch-project/k-NN/actions/runs/8871368904/job/24354253662#logs)

**What is the expected behavior?**
No failure

**What is your host/environment?**
Main branch and 2.x branch

**Do you have any screenshots?**
N/A

**Do you have any additional context?**
N/A

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -61,7 +61,7 @@ jobs:
             echo "avx2 not available on system"
             su `id -un 1000` -c "whoami && java -version && ./gradlew build -Dsimd.enabled=false"
           fi  
-          
+
 
       - name: Upload Coverage Report
         uses: codecov/codecov-action@v1
@@ -75,7 +75,7 @@ jobs:
 
     name: Build and Test k-NN Plugin on MacOS
     needs: Get-CI-Image-Tag
-    runs-on: macos-latest
+    runs-on: macos-12
 
     steps:
       - name: Checkout k-NN
@@ -92,39 +92,17 @@ jobs:
         with:
           java-version: ${{ matrix.java }}
 
-      - name: Build native libraries
+      - name: Install dependencies on macos
         run: |
-          git submodule update --init --recursive
           brew reinstall gcc
-          brew install llvm
-          brew install openblas
-          export FC=/opt/homebrew/bin/gfortran
-          export PATH=/opt/homebrew/opt/llvm/bin:$PATH
-          export CC=/opt/homebrew/opt/llvm/bin/clang
-          export CXX=/opt/homebrew/opt/llvm/bin/clang++
-          cd jni
-          sed -i -e 's/\/usr\/local\/opt\/libomp\//\/opt\/homebrew\/opt\/llvm\//g' cmake/init-faiss.cmake
-          sed -i -e 's/__aarch64__/__undefine_aarch64__/g' external/faiss/faiss/utils/distances_simd.cpp
-          sed -i -e 's/pragma message WARN/pragma message /g' external/nmslib/similarity_search/src/distcomp_scalar.cc
-          sed -i -e 's/-march=native/-mcpu=apple-m1/g' external/nmslib/similarity_search/CMakeLists.txt
-          sed -i -e 's/-mcpu=apple-a14/-mcpu=apple-m1/g' external/nmslib/python_bindings/setup.py
-          if sysctl -n machdep.cpu.features | grep -i AVX2;
-          then
-              echo "avx2 available on system"
-              cmake . --fresh
-          else
-              echo "avx2 not available on system"
-              cmake . --fresh -DSIMD_ENABLED=true
-          fi
-          cmake . --fresh
-          make
+          export FC=/usr/local/Cellar/gcc/12.2.0/bin/gfortran
 
       - name: Run build
         run: |
-          if sysctl -n machdep.cpu.features | grep -i AVX2;
+          if sysctl -n machdep.cpu.features machdep.cpu.leaf7_features | grep -i AVX2
           then
               echo "avx2 available on system"
-              ./gradlew build
+              ./gradlew build 
           else
               echo "avx2 not available on system"
               ./gradlew build -Dsimd.enabled=false

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -92,17 +92,39 @@ jobs:
         with:
           java-version: ${{ matrix.java }}
 
-      - name: Install dependencies on macos
+      - name: Build native libraries
         run: |
+          git submodule update --init --recursive
           brew reinstall gcc
-          export FC=/usr/local/Cellar/gcc/12.2.0/bin/gfortran
+          brew install llvm
+          brew install openblas
+          export FC=/opt/homebrew/bin/gfortran
+          export PATH=/opt/homebrew/opt/llvm/bin:$PATH
+          export CC=/opt/homebrew/opt/llvm/bin/clang
+          export CXX=/opt/homebrew/opt/llvm/bin/clang++
+          cd jni
+          sed -i -e 's/\/usr\/local\/opt\/libomp\//\/opt\/homebrew\/opt\/llvm\//g' cmake/init-faiss.cmake
+          sed -i -e 's/__aarch64__/__undefine_aarch64__/g' external/faiss/faiss/utils/distances_simd.cpp
+          sed -i -e 's/pragma message WARN/pragma message /g' external/nmslib/similarity_search/src/distcomp_scalar.cc
+          sed -i -e 's/-march=native/-mcpu=apple-m1/g' external/nmslib/similarity_search/CMakeLists.txt
+          sed -i -e 's/-mcpu=apple-a14/-mcpu=apple-m1/g' external/nmslib/python_bindings/setup.py
+          if sysctl -n machdep.cpu.features | grep -i AVX2;
+          then
+              echo "avx2 available on system"
+              cmake . --fresh
+          else
+              echo "avx2 not available on system"
+              cmake . --fresh -DSIMD_ENABLED=true
+          fi
+          cmake . --fresh
+          make
 
       - name: Run build
         run: |
-          if sysctl -n machdep.cpu.features machdep.cpu.leaf7_features | grep -i AVX2
+          if sysctl -n machdep.cpu.features | grep -i AVX2;
           then
               echo "avx2 available on system"
-              ./gradlew build 
+              ./gradlew build
           else
               echo "avx2 not available on system"
               ./gradlew build -Dsimd.enabled=false
@@ -166,4 +188,3 @@ jobs:
       - name: Run build
         run: |
           ./gradlew.bat build -D'simd.enabled=false'
-

